blocks/blockstore/write_cache.go

```diff
@@ -50,6 +54,8 @@ func (w *writecache) PutMany(bs []*blocks.Block) error {
 	for _, b := range bs {
 		if _, ok := w.cache.Get(b.Key()); !ok {
 			good = append(good, b)
+			k := b.Key();
+			defer log.EventBegin(context.TODO(), "Added block ", &k).Done()
```

Comment:
while it seems weird, it makes sense to me. We are registering the block to be stored here, but before the defer fires, the actual put will happen: `return w.blockstore.PutMany(good)`. So it should work as expected.